main.go

```diff
@@ -20,16 +21,55 @@ func main() {
 		defer f.Close()
 	}
 
-	err := logic.Setup()
+	parseErr := parse()
+	if parseErr != nil {
```

Comment:
```suggestion
	if parseErr := logic.Setup; parseErr != nil {
```

I suggest inlining the variable definition to make this code more readable and limit the variable's scope